src/server/hotspot_partition_calculator.h

```diff
@@ -17,13 +17,20 @@
 
 #pragma once
 
-#include "hotspot_partition_data.h"
+#include "hotspot_partition_stat.h"
 #include <gtest/gtest_prod.h>
 #include <dsn/perf_counter/perf_counter.h>
 
 namespace pegasus {
 namespace server {
 
+// stores the whole histories of all partitions in one table
+typedef std::list<std::vector<hotspot_partition_stat>> stat_histories;
+// hot_partition_counters c[index_of_partitions][type_of_read(0)/write(1)_stat]
+// so if we have n partitions, we will get 2*n hot_partition_counters, to demonstrate both
+// read/write hotspot value
+typedef std::vector<std::vector<std::unique_ptr<dsn::perf_counter_wrapper>>> hot_partition_counters;
```

Comment:
```suggestion
typedef std::vector<dsn::perf_counter_wrapper[2]> hot_partition_counters;
```
I think this is better. You don't need to wrap std::unique_ptr around perf_counter_wrapper. Because perf_counter_wrapper has only a pointer inside.

Reply:
It seems like a wrong syntax in C++

Reply:
```cpp
class perf_counter_wrapper
{
public:
    perf_counter_wrapper() { _counter = nullptr; }

    perf_counter_wrapper(const perf_counter_wrapper &other) = delete;
    perf_counter_wrapper(perf_counter_wrapper &other) = delete;
    perf_counter_wrapper(perf_counter_wrapper &&other) = delete;
    perf_counter_wrapper &operator=(const perf_counter_wrapper &other) = delete;
    perf_counter_wrapper &operator=(perf_counter_wrapper &other) = delete;
    perf_counter_wrapper &operator=(perf_counter_wrapper &&other) = delete;
```
It is hard to construct perf_counter_wrapper without unique_ptr in vector

Reply:
Use `std::array<dsn::perf_counter_wrapper, 2>` if you have syntax problem.

Reply:
Done